src/asm/rgbasm.5

```diff
@@ -0,0 +1,426 @@
+.\"
+.Dd April 9, 2017
+.Dt GBZ80 7
+.Os RGBDS Manual
+.Sh NAME
+.Nm rgbasm
+.Nd language documentation
+.Sh DESCRIPTION
+This is the full description of the language used by
+.Xr rgbasm 1 .
+The description of the instructions supported by the GameBoy CPU is in
+.Xr gbz80 7 .
+.Pp
+.Sh GENERAL
+.Ss Syntax
+The syntax line‐based, just as in any other assembler, meaning that you do one
+instruction or pseudo‐op per line:
+.Pp
+.Dl [label] [instruction] [;comment]
+.Pp
+Example:
+.Pp
+.Dl John: ld a,87 ;Weee
+.Pp
+All pseudo‐ops, mnemonics and registers (reserved keywords) are case‐insensitive
+and all labels are case‐sensitive.
+.Ss Sections
+Before you can start writing code, you must define a section.
+This tells the assembler what kind of information follows and, if it is code,
+where to put it.
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX
+.Pp
+This switches to the section called "CoolStuff" (or creates it if it doesn't
+already exist) and it defines it as a code section.
+All sections assembled at the same time that have the same name, type, etc, are
+considered to be the same one, and their code is put together in the object file
+generated by the assembler.
+All other sections must have a unique name, even in different source files, or
+the linker will treat it as an error.
+.Pp
+Possible section types are as follows:
+.Pp
+.Bl -tag
+.It Sy ROM0
+A ROM section.
+Mapped to memory at $0000–$3FFF (or $0000-$7FFF if tiny ROM mode is enabled in
+.Xr rgblink 1 ) .
+.It Sy ROMX
+A banked ROM section.
+Mapped to memory at $4000–$7FFF.
+Valid banks range from 1 to 511.
+Not available if tiny ROM mode is enabled in
+.Xr rgblink 1 .
+.It Sy VRAM
+A banked video RAM section.
+Mapped to memory at $8000–$9FFF.
+Can only allocate memory, not fill it.
+Valid banks are 0 and 1 but bank 1 isn't available if DMG mode is enabled in
+.Xr rgblink 1 .
+.It Sy SRAM
+A banked external (save) RAM section.
+Mapped to memory at $A000–$BFFF.
+Can only allocate memory, not fill it.
+Valid banks range from 0 to 15.
+.It Sy WRAM0
+A general-purpose RAM section.
+Mapped to memory at $C000–$CFFF, or $C000-$DFFF if DMG mode is enabled in
+.Xr rgblink 1.
+Can only allocate memory, not fill it.
+.It Sy WRAMX
+A banked general-purpose RAM section.
+Mapped to memory at $D000–$DFFF.
+Can only allocate memory, not fill it.
+Valid banks range from 1 to 7.
+Not available if DMG mode is enabled in
+.Xr rgblink 1.
+.It Sy OAM
+An object attributes RAM section.
+Mapped to memory at $FE00-$FE9F.
+Can only allocate memory, not fill it.
+.It Sy HRAM
+A high RAM section.
+Mapped to memory at $FF80–$FFFE.
+Can only allocate memory, not fill it.
+.Pp
+NOTE: If you use this method of allocating HRAM the assembler will NOT choose
+the short addressing mode in the LD instructions
+.Sy LD [$FF00+n8],A No and Sy LD A,[$FF00+n8] No because the actual address
+calculation is done by the linker.
+If you find this undesirable you can use
+.Ar RSSET No / Ar RB No / Ar RW No instead or use the
```

Comment:
Ar is for things that get replaced, not literal text. I would use Ic (internal command) for assembler directives.